### DISCUSSION TITLE
StatusChangeNotification

### DISCUSSION THREAD

**Discussion by ingdesarr1** [FreeOpcUa/opcua-asyncio#1600] [Q&A] [answered]
Good afternoon. I received the next StatusChangeNotification, but I dont know what it means. Can somebody help me? thanks

StatusChangeNotification(Status=StatusCode(value=2148270080), DiagnosticInfo_=DiagnosticInfo(SymbolicId=None, NamespaceURI=None, Locale=None, LocalizedText=None, AdditionalInfo=None, InnerStatusCode=None, InnerDiagnosticInfo=None))

This message is received when the PLC reset or recharge the program.

**Comment by AndreasHeine** [comment]
https://reference.opcfoundation.org/Core/Part4/v104/docs/7.20.1

The [NotificationMessage](https://reference.opcfoundation.org/search/17?t=NotificationMessage) structure used in the [Subscription](https://reference.opcfoundation.org/search/17?t=Subscription) [Service](https://reference.opcfoundation.org/search/17?t=Service) set allows specifying different types of [NotificationData](https://reference.opcfoundation.org/search/17?t=NotificationData). The [NotificationData](https://reference.opcfoundation.org/search/17?t=NotificationData) parameter is an extensible parameter whose structure depends on the type of [Notification](https://reference.opcfoundation.org/search/17?t=Notification) being sent. This parameter is defined in [Table 159](https://reference.opcfoundation.org/Core/Part4/v104/docs/?r=_Ref127332779). Other types can be defined by additional parts of series of standards or other specifications based on OPC UA. The [ExtensibleParameter](https://reference.opcfoundation.org/search/17?t=ExtensibleParameter) type is defined in [7.12](https://reference.opcfoundation.org/Core/Part4/v104/docs/?r=_Ref182070092).

There may be multiple notifications for a single [MonitoredItem](https://reference.opcfoundation.org/search/17?t=MonitoredItem) in a single NotificationData structure. When that happens the [Server](https://reference.opcfoundation.org/search/17?t=Server) shall ensure the notifications appear in the same order that they were queued in the [MonitoredItem](https://reference.opcfoundation.org/search/17?t=MonitoredItem). These notifications do not need to appear as a contiguous block.

https://reference.opcfoundation.org/Core/Part4/v104/docs/7.20.4

The [StatusChangeNotification](https://reference.opcfoundation.org/search/17?t=StatusChangeNotification) informs the [Client](https://reference.opcfoundation.org/search/17?t=Client) about a change in the status of a [Subscription](https://reference.opcfoundation.org/search/17?t=Subscription).

**Reply by ingdesarr1** [reply]
But how can I interpreter the code? is a int value, a hex value?

**Reply by AndreasHeine** [reply]
BadShutdown | 2148270080 | The operation was cancelled because the application is shutting down.

**Reply by AndreasHeine** [reply, marked as answer]
e.g. https://www.winccoa.com/documentation/WinCCOA/3.18/en_US/OPC_UA/opc_ua_c_status_codes.html

**Reply by AndreasHeine** [reply]
or -> https://github.com/FreeOpcUa/opcua-asyncio/blob/master/asyncua/ua/status_codes.py

**Reply by ingdesarr1** [reply]
Thanks

**Reply by ingdesarr1** [reply]
How can I get the message of the statuscode with the library?

**Reply by AndreasHeine** [reply]
there must be a method like read_data_value or so! the datavalue contains the statuscode.
